Raven.Client.Lightweight/Changes/RemoteChangesClientBase.cs

```diff
@@ -201,83 +213,149 @@ private bool DisposeHeartbeatTimer()
             }
         }
 
-        private Task lastSendTask;
-
-        protected Task Send(string command, string value)
+        async Task Worker()
         {
-            lock (this)
+            while (tokenSource.IsCancellationRequested == false)
             {
-                logger.Info("Sending command {0} - {1} to {2} with id {3}", command, value, url, id);
-                var sendTask = lastSendTask;
-                if (sendTask != null)
+                try
                 {
-                    return sendTask.ContinueWith(_ =>
-                    {
-                        Send(command, value);
-                    });
-                }
+                    await syncSource.Task.ConfigureAwait(false);
```

Comment:
Do we need to handle `TaskCancelledException` here?

Reply:
Ctor does

```
tokenRegistration = tokenSource.Token.Register(() => syncSource.TrySetCanceled());
```

which cancels the the task represented by the sync source. Would you much rather prefer to do

```
tokenRegistration = tokenSource.Token.Register(() => syncSource.TrySetResult());
```

instead and remove the catch?